src/App.tsx

```diff
@@ -1,223 +1,169 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  FormEvent,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { createTodo, getTodos, removeTodo } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/Auth/ErrorNotification/index';
+import { Footer } from './components/Auth/Footer/index';
+import { TodoList } from './components/Auth/TodoList/index';
+import { FilterStatus } from './types/FilterStatus';
+import { Todo } from './types/Todo';
+import { Error } from './types/Errors';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodos, setSelectedTodos] = useState<number[]>([]);
+  const [title, setTitle] = useState('');
+  const [error, setError] = useState<Error | null>(null);
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [isAdding, setIsAdding] = useState(false);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterStatus) {
+      case FilterStatus.All:
+        return todo;
+
+      case FilterStatus.Active:
+        return !todo.completed;
+
+      case FilterStatus.Completed:
+        return todo.completed;
+
+      default:
+        return null;
+    }
+  });
+
+  let userId = 0;
+
+  if (user?.id) {
+    userId = user.id;
+  }
+
+  if (error) {
+    setTimeout(() => {
+      setError(null);
+      setIsAdding(false);
+    }, 3000);
+  }
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
-    }
+    getTodos(userId)
+      .then(setTodos)
+      .catch(() => setError(Error.LOADING));
   }, []);
 
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setError(Error.ADDING);
+
+      return;
+    }
+
+    setIsAdding(true);
+
+    await createTodo(userId, title)
```

Comment:
you don't need `await` (and `async` above) here if used `then`
with `await` it should look in the following way:
```
const createdTodo = await createTodo(userId, title);
setTodos([...todos, todo]);
```
with `try-catch` block :arrow_up: